test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -88,4 +98,67 @@ static public void testPassingNullToIsFlattenedThrowsException() throws Throwabl
 			myUnsafe.isFlattened(null);
 		});
 	}
+
+	@Test
+	static public void testValueHeaderSizeOfValueType() throws Throwable {
+		long headerSize = myUnsafe.valueHeaderSize(ValueTypePoint2D.class);
+		if (isCompressedRefsEnabled) {
+			assertEquals(headerSize, 4);
+		} else {
+			assertEquals(headerSize, 8);
+		}
+	}
+
+	@Test
+	static public void testValueHeaderSizeOfPaddedValueType() throws Throwable {
+		long headerSize = myUnsafe.valueHeaderSize(ValueTypeWithLongField.class);
+		if (isCompressedRefsEnabled && !isFlatteningEnabled) {
+			assertEquals(headerSize, 4);
+		} else {
+			// the padded size is included in the header size, which will add an extra 4 bytes even when compressed refs are enabled
+			assertEquals(headerSize, 8);
+		}
+	}
+
+	// TODO (#14117): add testing for Unsafe.valueHeaderSize using Unsafe.getObjectSize
+
+	@Test
+	static public void testValueHeaderSizeOfNonValueType() throws Throwable {
+		assertEquals(myUnsafe.valueHeaderSize(Point2D.class), 0);
+	}
+
+	@Test
+	static public void testValueHeaderSizeOfNull() throws Throwable {
+		assertEquals(myUnsafe.valueHeaderSize(null), 0);
+	}
+
+	@Test
+	static public void testUninitializedDefaultValueOfValueType() throws Throwable {
+		// create a new ValueTypePoint2D to ensure that the class has been initialized befoere myUnsafe.uninitializedDefaultValue is called
+		ValueTypePoint2D newPoint = new ValueTypePoint2D(new ValueTypeInt(1), new ValueTypeInt(1));
+		ValueTypePoint2D p = myUnsafe.uninitializedDefaultValue(newPoint.getClass());
+		assertEquals(p.x.i, 0);
+		assertEquals(p.y.i, 0);
+	}
+
+	@Test
+	static public void testUninitializedDefaultValueOfNonValueType() throws Throwable {
+		Point2D p = myUnsafe.uninitializedDefaultValue(Point2D.class);
+		assertNull(p);
+	}
+
+	@Test
+	static public void testUninitializedDefaultValueOfNull() throws Throwable {
+		Object o = myUnsafe.uninitializedDefaultValue(null);
+		assertNull(o);
+	}
+
+	@Test
+	static public void testuninitializedVTClassHasNullDefaultValue() {
+		primitive class NeverInitialized {
+			final ValueTypeInt i;
+			NeverInitialized(ValueTypeInt i) { this.i = i; }
+		}
+		assertNull(myUnsafe.uninitializedDefaultValue(NeverInitialized.class));
+	}
```

Comment:
We can add test cases that pass array class to `valueHeaderSize()` and `uninitializedDefaultValue()`.

Reply:
addressed in d26e081